crates/noirc_evaluator/src/ssa/ssa_gen.rs

```diff
@@ -851,3 +767,237 @@ impl IRGenerator {
         Ok(v1.zip(&v2, &mut phi))
     }
 }
+
+impl IRGenerator {
+    /// Creates an ssa function and returns its type upon success
+    pub fn create_function(
```

Comment:
I'm in favor of moving these couple of functions here in the ssa_gen file so that all parts of the ssa_gen pass are within this file. If we think it is too large we can split it back up, though I think we should keep the ssa_gen parts separate from the other ssa pass-related parts of function.rs.

Regarding impls being in different files, I don't mind them. They are just functions with syntax sugar so I see no reason to disallow functions in other files.

Reply:
As noted on slack, I've reverted this change for now so we can push this PR through